src/NUnitCore/core/Compatibility.cs

```diff
@@ -0,0 +1,298 @@
+                        break;
+                    case "NUnit.Framework.TestFixtureTearDownAttribute":
+                        Error(location, "TestFixtureTearDownAttribute is not supported in NUnit 3. Use OneTimeTearDownAttribute.");
+                        break;
+                    case "NUnit.Framework.ValueSourceAttribute":
+                        sourceName = (string)Reflect.GetPropertyValue(attribute, "SourceName");
+                        if (!string.IsNullOrEmpty(sourceName))
+                        {
+                            Type sourceType = (Type)Reflect.GetPropertyValue(attribute, "SourceType");
+                            if (sourceType == null && parameterInfo != null)
+                                sourceType = parameterInfo.Member.ReflectedType;
+                            if (sourceType != null)
+                            {
+                                var members = sourceType.GetMember(sourceName, ALL_MEMBERS);
+                                if (members.Length > 0 && !IsStaticMember(members[0]))
+                                    Error(sourceType + "." + members[0].Name, "ValueSourceAttribute must reference a static member.");
+                            }
+                        }
+                        break;
+
+                    case "System.STAThreadAttribute":
+                    case "System.MTAThreadAttribute":
+                        Warning(location, attributeFullName + " has no effect in NUnit 3. Use ApartmentAttribute.");
+                        break;
+                }
+            }
+        }
+
+        #endregion
+
+        #region Helper Methods
+
+        private static bool IsStaticMember(MemberInfo member)
+        {
+            switch (member.MemberType)
+            {
+                case MemberTypes.Field:
+                    return ((FieldInfo)member).IsStatic;
+                case MemberTypes.Property:
+                    var getMethod = ((PropertyInfo)member).GetGetMethod();
+                    return getMethod != null && getMethod.IsStatic;
+                case MemberTypes.Method:
+                    return ((MethodInfo)member).IsStatic;
+                default:
+                    return false;
+            }
+        }
+
+        private static void Report(string level, string location, string message)
+        {
+            if (_writer != null)
+                _writer.WriteLine(string.Format(REPORT_FORMAT, level, location, message));
+        }
+
+        private static string LocationOf(MemberInfo member)
+        {
+            return member.ReflectedType + "." + member.Name;
+        }
+
+        #endregion
+
+        #region Nested Issue Class
+
+        public class Issue
+        {
+            public string Level { get; private set; }
+            public string Location { get; private set; }
+            public string Message { get; private set; }
```

Comment:
`private set;` signals that you're planning to mutate it from within the class, versus `{ get; }` which signals that the property is immutable.

Reply:
Unfortunately, just `{ get; }` doesn't work with the level of C# we use in the build. IIRC it's 4.0.

Reply:
I think I'll verify the level and do a PR to require it in each project.

Reply:
Setting LangVersion would be awesome. I switched to C# right around the time that C# 5 was released, so versions ≤ 5 run together in my mind.